components/application/widgets/CheckBoxButton.js

```diff
@@ -42,22 +50,33 @@ const CheckBoxButton = ({ priority, taskId, dueDate, projectId }) => {
       });
 
       const result = await res.json();
-      console.log(result);
-      result.ifComplete &&
-        dispatch(completeTaskAction({ taskId, dueDate, projectId }));
+      result.ifComplete && dispatch(completeTaskAction(taskId));
+      setIsCompleted(result.ifComplete);
     } catch (error) {
       throw error;
```

Comment:
I recommend using "throw new Error('XXX ' + error.message)" to facilitate future error localization.

Reply:
Fixed